src/pps/pps.go

```diff
@@ -0,0 +1,18 @@
+package pps
+
+import fmt "fmt"
+
+func (j *Job) String() string {
+	return fmt.Sprintf("%s@%s", j.Pipeline.Name, j.ID)
+}
+
+func (p *Pipeline) String() string {
+	if p == nil {
+		return "nil"
```

Comment:
I think we should just use `""` similar to a proto getter for a string field.

Reply:
The issue I see there is with something like:

```go
  var p *pps.Pipeline
  ⋮
  log.Println("processing pipeline", p)
```

The log entry would be `processing pipeline `, which is unlikely to be noticed; `processing pipeline nil` (or even something like `processing pipeline #<NIL>`) is a lot more likely to stick out as being wrong.

Reply:
yeah, I considered that, but I prefer the consistency for the return values (default values) when using this pattern rather than potentially better logs. Either way, I would like us to use this pattern less, and expect the protos to not be nil when calling these functions.